src/main.js

```diff
@@ -260,19 +261,42 @@ let load = base.loader({
         taskcluster: cfg.taskcluster,
         influx: influx,
         awsManager: awsManager,
-        provisionIterationInterval: cfg.app.iterationInterval,
-        dmsApiKey: cfg.deadmanssnitch.api.key,
-        iterationSnitch: cfg.deadmanssnitch.iterationSnitch,
+        //provisionIterationInterval: cfg.app.iterationInterval,
+        //dmsApiKey: cfg.deadmanssnitch.api.key,
+        //iterationSnitch: cfg.deadmanssnitch.iterationSnitch,
         stateContainer: stateContainer,
       });
 
-      try {
-        provisioner.run();
-      } catch (err) {
-        debug('[alert-operator] Error: ' + err.stack || err);
-      }
+      //return provisioner;
+      let i = new Iterate({
+        maxIterationTime: 1000 * 60 * 10,
+        watchDog: 1000 * 60 * 10,
+        maxFailures: 15,
+        waitTime: cfg.app.iterationInterval,
+        dmsConfig: {
+          apiKey: cfg.deadmanssnitch.api.key,
+          snitchUrl: cfg.deadmanssnitch.iterationSnitch,
+        },
+        handler: async (watchdog, state) => {
+          await provisioner.runAllProvisionersOnce();
+        },
+      });
+      i.start();
+      return new Promise((res, rej) => {
+        i.on('started', () => {
```

Comment:
i almost wonder if this shouldn't be stopped... that way you'd basically have a promise that lasts as long as the provisioner does, which might be nice.